eap-cluster-routing/src/main/java/com/relateiq/eapclusterrouting/util/EAPClusterRoutingCacheController.java

```
@@ -0,0 +1,94 @@
+
+import javax.inject.Singleton;
+import java.util.Optional;
+
+import static com.relateiq.eapclusterrouting.util.EAPClusterRoutingContants.EAPCLUSTERROUTING_CACHE_EXPIRATION_SEC;
+import static com.relateiq.eapclusterrouting.util.EAPClusterRoutingContants.EAPCLUSTERROUTING_CACHE_KEY_FORMAT;
+
+
+@Singleton
+public class EAPClusterRoutingCacheController {
+	private static final Logger log = LoggerFactory.getLogger(EAPClusterRoutingCacheController.class);
+
+	private final Gson gson;
+	private final RedisController redisController;
+
+	@Inject
+	public EAPClusterRoutingCacheController(RedisController redisController) {
+		this.redisController = redisController;
+		this.gson = new GsonBuilder()
+				.setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)
+				.create();
+	}
+
+	public Optional<EAPClusterRouting> getCachedEAPClusterRoutingUnsafe(String orgId) {
+		if (StringUtils.isNullOrEmpty(orgId)) {
+			log.error("orgId is empty");
+			return Optional.empty();
+		}
+
+		Optional<String> cacheKey = getRedisCacheKeyForOrgId(orgId);
+		if (!cacheKey.isPresent()) {
+			log.error("Unable to get the redis cache key.");
+			return Optional.empty();
+		}
+
+		final String ecrJson = redisController.get(cacheKey.get());
+		if (ecrJson != null) {
+			try {
+				final EAPClusterRouting eapClusterRouting = gson.fromJson(ecrJson, EAPClusterRouting.class);
+				return Optional.of(eapClusterRouting);
+			} catch (JsonSyntaxException e) {
+				log.error("Error deserializing cached EAPClusterRouting: {} ", e);
+			}
+		}
+		return Optional.empty();
+	}
+
+	public Long invalidateEAPClusterRoutingCache(String orgId) {
+		Optional<String> cacheKey = getRedisCacheKeyForOrgId(orgId);
+		if (!cacheKey.isPresent()) {
+			return 0L;
+		}
+
+		return redisController.del(cacheKey.get());
+	}
+
+	public void cacheEAPClusterRouting(EAPClusterRouting eapClusterRouting) {
+		if (eapClusterRouting == null || StringUtils.isNullOrEmpty(eapClusterRouting.getClusterId()) ||
```

nit: we can combine both **isNullOrEmpty** with **StringUtils.isAnyBlank**.
also: I think we should throw an IllegalArgumentException here instead of log and just return

---

Done.